fml/mapping.h

```diff
@@ -125,6 +125,44 @@ class NonOwnedMapping final : public Mapping {
   FML_DISALLOW_COPY_AND_ASSIGN(NonOwnedMapping);
 };
 
+/// A Mapping like NonOwnedMapping, but uses Free as its release proc.
+class MallocMapping final : public Mapping {
+ public:
+  MallocMapping();
+
+  MallocMapping(const uint8_t* data, size_t size);
+
+  MallocMapping(fml::MallocMapping&& mapping);
+
+  ~MallocMapping() override;
+
+  /// Copies the data from `begin` to `end`.
+  /// It's templated since void* arithemetic isn't allowed and we want support
+  /// for `uint8_t` and `char`.
+  template <typename T>
+  static MallocMapping Copy(const T* begin, const T* end) {
+    size_t length = end - begin;
+    return Copy(begin, length);
+  }
+
+  static MallocMapping Copy(const void* begin, size_t length);
```

Comment:
Docset for the signature. Specifically mentioning the following:
* The error handling to be expected. What happens on OOM and out of bounds access on the source buffer for instance.
* Clarify that size is in bytes.

Reply:
Suggestion: If this were `const uint8_t*`, the length parameter becomes much clearer.

Reply:
Done, I used void* in order to support uint8_t and int8_t.  Otherwise I'd have to use templates like above.